sphinx_gallery/gen_rst.py

```diff
@@ -49,9 +52,9 @@ def prefixed_lines():
         return ''.join(prefixed_lines())
 
 try:
-    from StringIO import StringIO
+    from BytesIO import BytesIO
 except ImportError:
-    from io import StringIO
+    from io import BytesIO
```

Comment:
I think this can be replaced by `from io import StringIO` if you are using the NonUnicodeFriendlyStringIO class below.

I wasn't aware of this but StringIO.StringIO doesn't have the same behaviour as io.StringIO (both exist in Python 2.7 so it's not a Python 2 vs Python 3 import difference as I naively thought). For example you can both write unicode and non unicode strings into it but then you get into trouble when you are trying to do `.getvalue`:

```python
from StringIO import StringIO
s.write('é')
s.write(u'é')
s.getvalue()
```

```
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-7-67a6f2fbafbe> in <module>()
----> 1 s.getvalue()

/volatile/le243287/miniconda3/envs/py27/lib/python2.7/StringIO.pyc in getvalue(self)
    269         _complain_ifclosed(self.closed)
    270         if self.buflist:
--> 271             self.buf += ''.join(self.buflist)
    272             self.buflist = []
    273         return self.buf

UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 0: ordinal not in range(128)
```